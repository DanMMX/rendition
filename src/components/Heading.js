import styled from 'styled-components'
import { color, fontSize, space } from 'styled-system'
import asRendition from '../asRendition'
import { monospace } from '../utils'
import { align, bold, caps } from './Txt'

const Heading = styled.h3`
  ${align}
  ${color}
  ${fontSize}
  ${monospace};
  ${space}

  ${caps}
  ${bold}
`

Heading.displayName = 'Heading'

Heading.defaultProps = {
  fontSize: 4,
  m: 0
}

const Base = asRendition(Heading)

Base.h1 = asRendition(Heading.withComponent('h1'))
Base.h1.defaultProps = {
  fontSize: 6,
  m: 0
}

Base.h2 = asRendition(Heading.withComponent('h2'))
Base.h2.defaultProps = {
  fontSize: 5,
  m: 0
}

Base.h3 = asRendition(Heading.withComponent('h3'))
Base.h3.defaultProps = {
  fontSize: 4,
  m: 0
}

Base.h4 = asRendition(Heading.withComponent('h4'))
Base.h4.defaultProps = {
  fontSize: 3,
  m: 0
}

Base.h5 = asRendition(Heading.withComponent('h5'))
Base.h5.defaultProps = {
  fontSize: 2,
  m: 0
}

Base.h6 = asRendition(Heading.withComponent('h6'))
Base.h6.defaultProps = {
  fontSize: 1,
  m: 0
}

export default Base
