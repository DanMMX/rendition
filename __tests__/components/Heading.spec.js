/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import Heading from '../../src/components/Heading'

test('Heading renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Heading>Heading</Heading>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
