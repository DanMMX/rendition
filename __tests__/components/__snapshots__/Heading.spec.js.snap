// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <h3
    className="sc-jTzLTM bUWCnN sc-VigVT eptWOS"
    fontSize={4}
  >
    Heading
  </h3>
</div>
`;
